Revit_Core_Engine/Create/View/View3D.cs

```diff
@@ -39,23 +41,31 @@ public static partial class Create
         [Input("viewName", "Optional, name of the new view.")]
         [Input("boundingBoxXyz", "Optional, the cuboid BoundingBoxXYZ to fit isometric view.")]
         [Input("viewTemplateId", "Optional, the View Template Id to be applied in the view.")]
-        [Input("viewDetailLevel", "Optional, the Detail Level of the view.")]        
+        [Input("viewDetailLevel", "Optional, the Detail Level of the view.")]
+        [Input("offset", "Offset that will be added to the BoundingBoxXYZ.")]
         [Output("view3D", "The new view.")]        
-        public static View View3D(this Document document, string viewName = null, BoundingBoxXYZ boundingBoxXyz = null, ElementId viewTemplateId = null, ViewDetailLevel viewDetailLevel = ViewDetailLevel.Coarse)
+        public static View View3D(this Document document, string viewName = null, BoundingBoxXYZ boundingBoxXyz = null, ElementId viewTemplateId = null, ViewDetailLevel viewDetailLevel = ViewDetailLevel.Coarse, double offset = 0)
         {
-            View result = null;
-
+            View newView = null;
             ViewFamilyType vft = Query.ViewFamilyType(document, ViewFamily.ThreeDimensional);
 
-            result = Autodesk.Revit.DB.View3D.CreateIsometric(document, vft.Id);
-            
-            Modify.SetViewDetailLevel(result, viewDetailLevel);
+            newView = Autodesk.Revit.DB.View3D.CreateIsometric(document, vft.Id);
+
+            if (viewDetailLevel != ViewDetailLevel.Undefined)
+                Modify.SetViewDetailLevel(newView, viewDetailLevel);
 
             if (boundingBoxXyz != null)
             {
+                if (offset != 0)
```

Comment:
It is always safer to use some tolerance
```C#
                if (Math.Abs(offset) > tolerance)
```

Reply:
Safety first 👍